lib/iris/fileformats/rules.py

```diff
@@ -935,60 +935,113 @@ def _make_cube(field, converter):
     return cube, metadata.factories, metadata.references
 
 
-def load_cubes(filenames, user_callback, loader, filter_function=None):
+def _resolve_factory_references(cube, factories, concrete_reference_targets,
+                                regrid_cache={}):
+    # Attach the factories for a cube, building them from references.
+    # Note: the regrid_cache argument lets us share and reuse regridded data
+    # across multiple result cubes.
+    for factory in factories:
+        try:
+            args = _dereference_args(factory, concrete_reference_targets,
+                                     regrid_cache, cube)
+        except _ReferenceError as e:
+            msg = 'Unable to create instance of {factory}. ' + str(e)
+            factory_name = factory.factory_class.__name__
+            warnings.warn(msg.format(factory=factory_name))
+        else:
+            aux_factory = factory.factory_class(*args)
+            cube.add_aux_factory(aux_factory)
+
+
+def _load_pairs_from_fields_and_filenames(fields_and_filenames, converter,
+                                          user_callback=None):
+    # The underlying mechanism for the public 'load_pairs_from_fields' and
+    # 'load_cubes'.
+    # Slightly more complicated than 'load_pairs_from_fields', only because it
+    # needs a filename associated with each field to support the load callback.
     concrete_reference_targets = {}
     results_needing_reference = []
+    for field, filename in fields_and_filenames:
+        # Convert the field to a Cube, passing down the 'converter' function.
+        cube, factories, references = _make_cube(field, converter)
+
+        # Post modify the new cube with a user-callback.
+        # This an ordinary Iris load callback, so it takes the filename.
+        cube = iris.io.run_callback(user_callback, cube, field, filename)
```

Comment:
**Sorry I seem to have missed an important point here.**
Comparing this with the code in the original 'load_cubes', I think we must skip the rest of the loop (ie.c. "continue"), if 'cube' is None.

This is far from completely clear IMHO.
The description in "iris.io.run_callback" says
 * "If the cube is to be rejected the callback must raise an iris.exceptions.IgnoreCubeException.", and 
 * "It is possible that this function returns None for certain callbacks, the caller of this function should handle this case."

However, the usage in the other file loaders (including the original of this) seems clear : you should skip it .